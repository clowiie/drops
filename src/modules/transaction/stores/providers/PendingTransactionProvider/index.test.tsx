import { render } from '@testing-library/react'

import * as PendingTransactionProviderType from '.'

describe('PendingTransactionProvider', () => {
  const usePendingTransactionSpy = jest.fn()
  jest.doMock(
    'modules/transaction/hooks/usePendingTransaction',
    () => usePendingTransactionSpy,
  )

  const Component = () => <h1>Component</h1>

  const { default: PendingTransactionProvider } =
    require('.') as typeof PendingTransactionProviderType

  beforeEach(() => {
    usePendingTransactionSpy.mockReturnValue({
      pendingHashes: ['hash1', 'hash2'],
      hasPendingHash: true,
      setPendingHashes: () => {},
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container, getByText } = render(
        <PendingTransactionProvider>
          <Component />
        </PendingTransactionProvider>,
      )

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Component')).toBeDefined()
    })
  })
})
