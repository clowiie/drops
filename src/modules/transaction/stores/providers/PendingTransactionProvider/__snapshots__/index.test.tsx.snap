// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PendingTransactionProvider snapshots should match snapshots 1`] = `
<h1>
  Component
</h1>
`;
